fix(docs): clarify database migrations (#13026)
getsentry/sentry-docs: develop-docs/backend/application-domains/database-migrations/index.mdx

To avoid this, follow these steps:

> - First, if the column is either not nullable, or doesn't have a `db_default` set, then make a PR to make it nullable via `null=True`.

Reviewer: Right, that makes sense to me!

> - First, if the column is either not nullable, or doesn't have a `db_default` set. If either of these is true, then make it nullable via `null=True`.

Reviewer 2: ```suggestion
- First, if the column is either not nullable, or doesn't have a `db_default` set, then make a PR to make it nullable via `null=True`.
```

> - Then, make a PR to remove all uses of the column in the codebase in a separate PR. This mostly helps with code cleanliness. This should be merged ahead of the next migration PRs, but we don't need to worry about whether it is deployed first.

Reviewer 3: ```suggestion
- Then, remove all uses of the column in the codebase in a separate PR; this mostly helps with code cleanliness. This should be merged ahead of the next migration PRs, but we don't need to worry about whether it is deployed first.
```
